vector/v.distance/main.c

```diff
@@ -1504,35 +1522,51 @@ int main(int argc, char *argv[])
         }
 
         if (print) { /* print only */
-            /*
-               input and output is the same &&
-               calculate distances &&
-               only one upload option given ->
-               print as a matrix
-             */
-            if (print_as_matrix) {
-                if (i == 0) {
-                    for (j = 0; j < nfrom; j++) {
-                        if (j == 0)
-                            fprintf(stdout, " ");
-                        fprintf(stdout, "%s%d", sep, Near[j].to_cat);
+            switch (format) {
+            case PLAIN:
+                /*
+                   input and output is the same &&
+                   calculate distances &&
+                   only one upload option given ->
+                   print as a matrix
+                 */
+                if (print_as_matrix) {
+                    if (i == 0) {
+                        for (j = 0; j < nfrom; j++) {
+                            if (j == 0)
+                                fprintf(stdout, " ");
+                            fprintf(stdout, "%s%d", sep, Near[j].to_cat);
+                        }
+                        fprintf(stdout, "\n");
+                    }
+                    if (i % nfrom == 0) {
+                        fprintf(stdout, "%d", Near[i].from_cat);
+                        for (j = 0; j < nfrom; j++) {
+                            print_upload(Near, Upload, i + j, &cvarr, catval,
+                                         sep, format, NULL);
+                        }
+                        fprintf(stdout, "\n");
                     }
-                    fprintf(stdout, "\n");
                 }
-                if (i % nfrom == 0) {
+                else {
                     fprintf(stdout, "%d", Near[i].from_cat);
-                    for (j = 0; j < nfrom; j++) {
-                        print_upload(Near, Upload, i + j, &cvarr, catval, sep);
-                    }
+                    if (do_all)
+                        fprintf(stdout, "%s%d", sep, Near[i].to_cat);
+                    print_upload(Near, Upload, i, &cvarr, catval, sep, format,
+                                 NULL);
                     fprintf(stdout, "\n");
                 }
-            }
-            else {
-                fprintf(stdout, "%d", Near[i].from_cat);
-                if (do_all)
-                    fprintf(stdout, "%s%d", sep, Near[i].to_cat);
-                print_upload(Near, Upload, i, &cvarr, catval, sep);
-                fprintf(stdout, "\n");
+                break;
+            case JSON:
```

Comment:
Let's add a json representation for print as matrix `-s` flag. Maybe something like this schema, but I'm open to suggestions.

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "DistanceMatrix",
  "description": "A matrix representing distances between points",
  "type": "array",
  "items": {
    "type": "array",
    "items": {
      "type": "number"
    }
  }
}
```